detect_secrets/plugins/aws.py

```diff
@@ -24,6 +24,7 @@ class AWSKeyDetector(RegexBasedDetector):
 
     denylist = (
         re.compile(r'AKIA[0-9A-Z]{16}'),
+        re.compile(r'aws(.{0,20})?[\'\"][0-9a-zA-Z/+]{40}[\'\"]'),
```

Comment:
Actually @pablosantiagolopez , do you have an example of the type of secrets that this should catch?

Reply:
Yes, some examples that match with this regex could be:
- `aws_secretorotherkey = "SecretValueWithExactly40AlphaNumberChars"`: this secret will be detected by `KeywordDetector` plugin (with the improvements implemented in #396) and with the `AWSKeyDetector` plugin.
- `aws_personal = "SecretValueWithExactly40AlphaNumberChars"`: this secret only will be detected by the `AWSKeyDetector` plugin.

Reply:
Interesting. It looks like you're suggesting that this examines the variable name, rather than the secret value.

Could you write a test case that formalizes this? Additionally, could you test that the detected `secret_value` is indeed what you expect it to be? As this regex has more capturing groups than typically expected, I wonder how it will be handled by the rest of the system (including `audit` functionality).